adcm_aio_client/core/objects/cm.py

```diff
@@ -81,6 +86,32 @@ def get_own_path(self: Self) -> Endpoint:
 class BundlesNode(PaginatedAccessor[Bundle, None]):
     class_type = Bundle
 
+    async def _download_external_bundle(self: Self, url: str) -> bytes:
+        try:
+            urlopen(url)  # noqa S310
+            async with httpx.AsyncClient() as client:
+                response = await client.get(url)
+                return response.content
+        except ValueError as err:
+            raise OperationError from err
+
+    async def create(self: Self, bundle_loc: str, accept_license: bool | None = None) -> Bundle:
+        if not Path(bundle_loc).exists():
+            file_content = await self._download_external_bundle(bundle_loc)
+            files = {"file": file_content}
+        else:
+            files = {"file": Path(bundle_loc).read_bytes()}
+
+        response = await self._requester.post("bundles", data=files)
+
+        if accept_license:
+            (await self.get()).license.accept()
```

Comment:
build bundle before that, then accept license

you can already have bundle object at this point

Reply:
changed